.github/workflows/ci.yml

```diff
@@ -0,0 +1,75 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - "*.*.*"
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+            ~/.coursier
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-${{ matrix.scala_version }}
+          restore-keys: |
+            ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-
+      - name: Build
+        run: sbt compile packageBin packageDoc packageSrc
+      - name: Test
+        run: sbt test
+      - uses: actions/upload-artifact@v2
+        with:
+          name: target
+          path: '*/target/'
+          retention-days: 1
+  publish:
+    if: github.event_name != 'pull_request'
+    needs: build-and-test
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+            ~/.coursier
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-${{ matrix.scala_version }}
+          restore-keys: |
+            ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-
+      - uses: actions/download-artifact@v2
+        with:
+          name: target
+      - name: Publish to Sonatype
+        run: bash publish.sh
+        env:
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
+  release:
+    if: startsWith(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
```

Comment:
is using ubuntu-latest instead of ubuntu-20.04 for just this job intentional?

Reply:
eh, sort of. I don't remember my exact thoughts at the time, but I'm less worried about an ubuntu upgrade braking this job than I am the others.